File: python/setup_statusline.py
import json
import os
import sys

SETTINGS_PATH = os.path.join(os.path.expanduser("~"), ".claude", "settings.json")

# absolute path to meter.py next to this script
METER_PATH = os.path.join(os.path.dirname(os.path.abspath(__file__)), "meter.py")

REFRESH_INTERVAL = 5000


def load_settings():
    if not os.path.exists(SETTINGS_PATH):
        return {}
    try:
        with open(SETTINGS_PATH, "r") as f:
            return json.load(f)
    except (json.JSONDecodeError, IOError):
        return {}


def save_settings(settings):
    os.makedirs(os.path.dirname(SETTINGS_PATH), exist_ok=True)
    with open(SETTINGS_PATH, "w") as f:
        json.dump(settings, f, indent=2)
        f.write("\n")


def desired_command():
    return f'python "{METER_PATH}"'


def needs_update(settings):
    sl = settings.get("statusLine", {})
    return sl.get("command") != desired_command()


def main():
    settings = load_settings()

    if not needs_update(settings):
        # already configured, nothing to do
        result = {"additionalContext": "chonker context meter is active on status line"}
        print(json.dumps(result))
        return

    settings["statusLine"] = {
        "type": "command",
        "command": desired_command(),
        "refreshInterval": REFRESH_INTERVAL,
    }
    save_settings(settings)

    result = {
        "additionalContext": "chonker context meter installed and active on status line"
    }
    print(json.dumps(result))


if __name__ == "__main__":
    main()

File: hooks/setup_statusline.js
const fs = require("fs");
const path = require("path");
const os = require("os");

const SETTINGS_PATH = path.join(os.homedir(), ".claude", "settings.json");

// absolute path to meter.js next to this script
const METER_PATH = path.join(__dirname, "meter.js");

const REFRESH_INTERVAL = 5000;

function loadSettings() {
  try {
    return JSON.parse(fs.readFileSync(SETTINGS_PATH, "utf-8"));
  } catch {
    return {};
  }
}

function saveSettings(settings) {
  fs.mkdirSync(path.dirname(SETTINGS_PATH), { recursive: true });
  fs.writeFileSync(SETTINGS_PATH, JSON.stringify(settings, null, 2) + "\n");
}

function desiredCommand() {
  return `"${process.execPath}" "${METER_PATH}"`;
}

function needsUpdate(settings) {
  const sl = settings.statusLine || {};
  return sl.command !== desiredCommand();
}

function main() {
  const settings = loadSettings();

  if (!needsUpdate(settings)) {
    // already configured, nothing to do
    console.log(
      JSON.stringify({
        additionalContext: "chonker context meter is active on status line",
      })
    );
    return;
  }

  settings.statusLine = {
    type: "command",
    command: desiredCommand(),
    refreshInterval: REFRESH_INTERVAL,
  };
  saveSettings(settings);

  console.log(
    JSON.stringify({
      additionalContext:
        "chonker context meter installed and active on status line",
    })
  );
}

main();
